marathon/src/test/scala/io/buoyant/marathon/v2/AppIdNamerTest.scala

```diff
@@ -114,4 +134,138 @@ class AppIdNamerTest extends FunSuite with Awaits {
       assert(addrs == addresses)
     }
   }
+
+  test("Namer recovers if marathon api fails initially") {
+    val api = TestApi(
+      ids = Future.value(Set(Path.read("/foo/bar"))),
+      addrs = Future.never,
+      initIdsAlive = false,
+      initAddrsAlive = true
+    )
+
+    val namer = new AppIdNamer(api, Path.Utf8("io.l5d.marathon"), ttl)
+    @volatile var state: Activity.State[NameTree[Name]] = Activity.Pending
+
+    val input = Path.Utf8("foo", "bar", "residual")
+    val output = Path.Utf8("io.l5d.marathon", "foo", "bar")
+    namer.lookup(input).states.respond(state = _)
+    assert(state == Activity.Failed(err))
+
+    // wait for marathon api to come back
+    api.idsAlive = true
+    Thread.sleep(ttl.inMillis * 3)
```

Comment:
To recap IRL discussion: we should try to replace these sleeps with use of a MockTimer so that these tests are faster and less prone to flakiness